.evergreen/config.yml

```diff
@@ -518,6 +518,47 @@ functions:
           PKG_CONFIG_PATH=$PKG_CONFIG_PATH \
           LD_LIBRARY_PATH=$LD_LIBRARY_PATH
 
+  run-load-balancer-tests:
+    - command: shell.exec
+      type: test
+      params:
+        working_dir: src/go.mongodb.org/mongo-driver
+        script: |
+          ${PREPARE_SHELL}
+
+          if [ ${SSL} = "ssl" ]; then
+              export MONGO_GO_DRIVER_CA_FILE="$PROJECT_DIRECTORY/data/certificates/ca.pem"
+              if [ "Windows_NT" = "$OS" ]; then # Magic variable in cygwin
+                  export MONGO_GO_DRIVER_CA_FILE=$(cygpath -m $MONGO_GO_DRIVER_CA_FILE)
+              fi
+          fi
+
+          # Verify that the required LB URI env vars are set to ensure that the test runner can correctly connect to
```

Comment:
We check for empty LB URIs both here and in `mongo/integration/mtest/setup.go`. This might be redundant, but it accounts for both running in CI and locally. If we only checked in `setup.go`, we could have an issue where `SINGLE_MONGOS_LB_URI` is unset, so `MONGODB_URI` would also be unset. In this case, the `setup` function defaults to `localhost:27017` and based on topology discovery, thinks it's connecting to a mongos rather than an LB cluster, so important tests that only run against LBs would be incorrectly skipped. Having the check in both places prevents that from happening.

Note that this may not be required once an upstream service fully supports running behind an LB because the service should error if it's configured in LB mode but the driver connects without `loadBalanced=true`.

Reply:
Nice, the extra check to avoid accidentally skipping these tests is a good precaution.

> Note that this may not be required once an upstream service fully supports running behind an LB because the service should error if it's configured in LB mode but the driver connects without loadBalanced=true.

Right, since drivers send `loadBalanced: true` as part of the `hello` command, the server could reject non load balanced connections if it knows it is running behind a load balancer.